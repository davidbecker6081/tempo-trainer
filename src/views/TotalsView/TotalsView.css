.totals-view {
  flex-direction: column;
  padding: 2em;
  width: 60vw;
  background-color: #303032;
  border-radius: 5px;
}

.totals-wrapper {
  width: 100%;
}

@media screen and (max-width: 770px) {
  .totals-wrapper {
    flex-direction: column;
  }
}
