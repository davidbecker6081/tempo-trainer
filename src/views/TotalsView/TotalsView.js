import React from 'react';
import PropTypes from 'prop-types';
import Total from '../../components/Total/Total';
import './TotalsView.css';

const TotalsView = ({ dataHelper }) =>
  (
    <section>
      <Total dataHelper={dataHelper} channelSet="millisecondOffset" />
      <Total dataHelper={dataHelper} channelSet="distance" />
      <Total dataHelper={dataHelper} channelSet="elevation" />
    </section>
  );

export default TotalsView;

TotalsView.propTypes = {
  dataHelper: PropTypes.shape({
    data: PropTypes.shape({}),
    channels: PropTypes.arrayOf(PropTypes.string),
    GPSCoords: PropTypes.arrayOf(PropTypes.shape({})),
  }).isRequired,
};
