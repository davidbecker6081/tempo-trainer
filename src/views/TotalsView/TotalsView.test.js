import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import TotalsView from './TotalsView';
import DataHelper from '../../components/DataCleaner/DataCleaner';
import workoutData from '../../__mock__/workout-data.json';

describe('TotalsView', () => {
  let wrapper;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      dataHelper: {},
    };
  });

  describe('dataHelper is defined', () => {
    beforeEach(() => { mockProps.dataHelper = new DataHelper(workoutData); });
    it('should render correclty', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
