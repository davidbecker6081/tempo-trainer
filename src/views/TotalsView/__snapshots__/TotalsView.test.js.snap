// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TotalsView dataHelper is defined should render correclty 1`] = `undefined`;
